ledger-service/jwt/src/main/scala/com/digitalasset/jwt/JwtVerifier.scala

```diff
@@ -43,31 +43,74 @@ object JwtVerifier {
   }
 }
 
+final case class LeewayOptions(
+    leeway: Option[Long] = None,
+    expiresAt: Option[Long] = None,
+    issuedAt: Option[Long] = None,
+    notBefore: Option[Long] = None,
+)
+
+sealed trait Leeway {
+  def getVerifier(
+      algorithm: Algorithm,
+      mbLeewayOptions: Option[LeewayOptions] = None,
+  ): com.auth0.jwt.interfaces.JWTVerifier = {
+    def addLeeway(verification: Verification, leewayOptions: LeewayOptions): Verification = {
+      def appendVerification(option: Option[Long], f: (Verification, Long) => Verification)(
+          verifier: Verification
+      ): Verification =
+        option match {
+          case None => verifier
+          case Some(value) => f(verifier, value)
+        }
+      val mbOptionsActions: List[(Option[Long], (Verification, Long) => Verification)] = List(
+        (leewayOptions.leeway, _.acceptLeeway(_)),
+        (leewayOptions.expiresAt, _.acceptExpiresAt(_)),
+        (leewayOptions.issuedAt, _.acceptIssuedAt(_)),
+        (leewayOptions.notBefore, _.acceptNotBefore(_)),
+      )
+      mbOptionsActions.foldLeft(verification)((v, elem) => {
```

Comment:
you could decompose `elem` using pattern matching here, i.e. smth like:

```
mbOptionsActions.foldLeft(verification) {
        case (verifier, (None, _)) => verifier
        case (verifier, (Some(value), f)) => f(verifier, value)
      }
```

Reply:
please note I removed the need of `appendVerification` method as well

Reply:
Yes it is more readable now.